packages/language-server/src/test/prisma-fmt-referentialActions.test.ts

```diff
@@ -0,0 +1,192 @@
+            User   User?        @relation(fields: [userId], references: [id])
+            userId Int?
+        }
+        
+        model TagOnPosts {
+            id     Int   @id @default(autoincrement())
+            post   Post? @relation(fields: [postId], references: [id], onUpdate: Cascade, onDelete: Cascade)
+            tag    Tag?  @relation(fields: [tagId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+            postId Int?
+            tagId  Int?
+        }
+        
+        model Tag {
+            id    Int          @id @default(autoincrement())
+            name  String       @unique
+            posts TagOnPosts[]
+        }`
+
+    const actions = referentialActions(binPath, schema)
+
+    assert.deepStrictEqual(actions, [
+      'Cascade',
+      'Restrict',
+      'NoAction',
+      'SetNull',
+      'SetDefault',
+    ])
+  })
+
+  test('MySQL', async () => {
+    const binPath = await util.getBinPath()
+
+    const schema = `
+        datasource db {
+            provider = "mysql"
+            url      = env("DATABASE_URL")
+        }
+        
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["referentialActions"]
+        }`
+
+    const actions = referentialActions(binPath, schema)
+
+    assert.deepStrictEqual(actions, [
+      'Cascade',
+      'Restrict',
+      'NoAction',
+      'SetNull',
+      'SetDefault',
+    ])
+  })
+
+  test('SQL Server', async () => {
+    const binPath = await util.getBinPath()
+
+    const schema = `
+        datasource db {
+            provider = "sqlserver"
+            url      = env("DATABASE_URL")
+        }
+        
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["microsoftSqlServer", "referentialActions"]
+        }`
+
+    const actions = referentialActions(binPath, schema)
+
+    assert.deepStrictEqual(actions, [
+      'Cascade',
+      'NoAction',
+      'SetNull',
+      'SetDefault',
+    ])
+  })
+
+  test('no datasource should return empty []', async () => {
+    const binPath = await util.getBinPath()
+
+    const schema = `
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["referentialActions"]
+        }`
+
+    const actions = referentialActions(binPath, schema)
+
+    assert.deepStrictEqual(actions, [])
+  })
+
+  test('invalid schema should return empty []', async () => {
+    const binPath = await util.getBinPath()
+
+    const schema = `
+        datasource db {
+            provider = "sqlite"
+            url      = env("DATABASE_URL")
+        }
+
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["referentialActions"]
+        }
+        
+        model { sss } // invalid model
+        `
+
+    const actions = referentialActions(binPath, schema)
+
+    assert.deepStrictEqual(actions, [])
+  })
```

Comment:
_Should_ it really return an empty array or is that just what it does right now? I would probably expect it to return all options by default so the user can do _something - and when they ad more information to their schema (or fix it), the suggestions would be improved (and old autocompleted items fixed via validation).

Reply:
It always returns an array either empty or not right now. It's not optimal.